firmware/controllers/trigger/trigger_emulator_algo.cpp

```diff
@@ -106,7 +106,7 @@ static void updateTriggerWaveformIfNeeded(PwmConfig *state DECLARE_ENGINE_PARAME
 
 
 		TriggerWaveform *s = &engine->triggerCentral.triggerShape;
-		copyPwmParameters(state, s->getSize(), &s->wave);
+		copyPwmParameters(state, &*s->wave);
```

Comment:
address-of-dereference-of?

Reply:
s->wave is a MultiChannelStateSequenceWithData, which is really just a uint8_t[some_size];
It does not derive from MultiChannelStateSequence.  So in order to access the Multi*Sequence, there are four overloaded operators: * and ->, both const and non-const versions.

So really the * is saying "give a reference the object you're managing" and & is saying "give me a pointer to it", since there is no operator that says give me the pointer.

Though maybe I could add
operator MultiChannelStateSequenceWithData*()
?